releasetool/commands/start/java.py

```diff
@@ -0,0 +1,229 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import getpass
+import os
+import textwrap
+from typing import Collection, Optional, Sequence
+from glob import glob
+
+import attr
+import click
+import re
+
+import releasetool.filehelpers
+import releasetool.git
+import releasetool.github
+import releasetool.secrets
+import releasetool.commands.common
+
+
+@attr.s(auto_attribs=True, slots=True)
+class Context(releasetool.commands.common.GitHubContext):
+    package_name: Optional[str] = None
+    last_release_version: Optional[str] = None
+    last_release_committish: Optional[str] = None
+    changes: Sequence[str] = ()
+    pom_files: Collection[str] = []
+    snapshot_version: Optional[str] = None
+    release_notes: Optional[str] = None
+    release_version: Optional[str] = None
+    release_branch: Optional[str] = None
+    pull_request: Optional[dict] = None
+
+
+def determine_package_name(ctx: Context) -> None:
+    click.secho("> Figuring out the package name.", fg="cyan")
+    ctx.package_name = os.path.basename(os.getcwd())
+    click.secho(f"Looks like we're releasing {ctx.package_name}.")
+
+
+def determine_last_release(ctx: Context) -> None:
+    click.secho("> Figuring out what the last release was.", fg="cyan")
+    tags = releasetool.git.list_tags()
+    candidates = [tag for tag in tags if tag.startswith("v")]
+
+    if candidates:
+        ctx.last_release_committish = candidates[0]
+        # strip the leading 'v'
+        ctx.last_release_version = candidates[0].lstrip("v")
+
+    else:
+        click.secho(
+            f"I couldn't figure out the last release for {ctx.package_name}, "
+            "so I'm assuming this is the first release. Can you tell me "
+            "which git rev/sha to start the changelog at?",
+            fg="yellow",
+        )
+        ctx.last_release_committish = click.prompt("Committish")
+        ctx.last_release_version = "0.0.0"
+
+    click.secho(f"The last release was {ctx.last_release_version}.")
+
+
+def determine_snapshot_version(ctx: Context) -> None:
+    click.secho("> Figuring out the current snapshot version.", fg="cyan")
+
+    with open("pom.xml", "r") as fh:
+        content = fh.read()
+        m = re.search(r'<version>(\d+\.\d+\.\d+)-SNAPSHOT</version>', content)
+        if m:
+            ctx.snapshot_version = m.group(1)
+
+    if ctx.snapshot_version is None:
+        click.secho(
+            "I couldn't figure out the current snapshot version from pom.xml.",
+            fg="red",
+        )
+        click.exit(1)
+
+
+def gather_changes(ctx: Context) -> None:
+    click.secho(f"> Gathering changes since {ctx.last_release_version}", fg="cyan")
+    ctx.changes = releasetool.git.summary_log(ctx.last_release_committish)
```

Comment:
Can you update this to match https://github.com/GoogleCloudPlatform/releasetool/pull/30/files#diff-e008eecec979f583f499eb3a367a8d12R90

Reply:
done.